native/src/components/DatesPageDetail.tsx

```diff
@@ -24,35 +38,44 @@ type DatesPageDetailProps = {
 }
 
 const DatesPageDetail = ({ date, languageCode }: DatesPageDetailProps): ReactElement | null => {
-  const dates = date.recurrences(MAX_DATE_RECURRENCES).map(it => it.toFormattedString(languageCode, true))
-  const nextDate = dates[0] ?? date.toFormattedString(languageCode, true)
-  const hasMoreDates = date.hasMoreRecurrencesThan(MAX_DATE_RECURRENCES)
-  const { t } = useTranslation('events')
+  const [tapsOnShowMore, setTapsOnShowMore] = useState(0)
+  const visibleRecurrences = MAX_DATE_RECURRENCES * (tapsOnShowMore + 1)
 
-  if (dates.length === 1) {
-    return <PageDetail identifier={t('date_one')} information={nextDate} language={languageCode} />
-  }
+  // Use the content language to match the surrounding translations
+  const { t: translateIntoContentLanguage } = useTranslation('events', { lng: languageCode })
 
-  const Title = <Identifier>{t(hasMoreDates ? 'nextDate_other' : 'date_other')}: </Identifier>
-  const Dates = dates.map(it => <StyledText key={it}>{it}</StyledText>)
-  const AlwaysShownDates = <>{Dates.slice(0, MAX_DATE_RECURRENCES_COLLAPSED)}</>
+  const recurrences = date
+    .recurrences(MAX_DATE_RECURRENCES * (tapsOnShowMore + 1))
+    .map(recurrence => recurrence.formatMonthlyOrYearlyRecurrence(languageCode, translateIntoContentLanguage))
+    .map(formattedDate => (
+      <SingleDateContainer key={formattedDate.date}>
+        <PageDetail Icon={CalendarTodayIcon} information={formattedDate.date} language={languageCode} />
+        <PageDetail Icon={ClockIcon} information={formattedDate.time} language={languageCode} />
+      </SingleDateContainer>
+    ))
```

Comment:
❓ This looks expensive... do we need to use useMemo if it recalculate on each re-render?

Reply:
Hmm, I'm not sure that would help here since the recurrences already change on the first part, since a re-render would be mostly because of additional taps, so we'd have an additional three recurrences going through the mapping.

But it's also not that expensive, there is no grand re-sorting or anything like that. I tested it in native and web up to some 50 results and didn't notice any slowing down.